File: Prototype/BigGp/src/Model/BaseNodes/ForStatement/ForAssignment.java
package Model.BaseNodes.ForStatement;

import Model.Node;
import Model.SmallNodes.Expressions.Standart.Childs.VarExpression;
import Model.SmallNodes.Expressions.Standart.Expression;
import Serialization.Token;

import java.util.ArrayList;

public class ForAssignment extends Node {

    private VarExpression variable;
    private Expression expression;

    public VarExpression getVariable() {
        return variable;
    }

    @Override
    public String toString() {
        return variable.toString() + " = " + expression.toString();
    }

    public void setExpression(Expression expression) {
        this.expression = expression;
    }

    public ArrayList<Node> getChildrenAsNodes(){
        ArrayList<Node> nodes = new ArrayList<>();
        nodes.add(this);
        nodes.addAll(variable.getChildrenAsNodes());
        nodes.addAll(expression.getChildrenAsNodes());

        return nodes;
    }

    public ForAssignment(Node parentNode) {
        super(parentNode, "ForAssignment", true);
        this.expression = new Expression(this);
        this.variable = new VarExpression(this);
    }

    public ForAssignment(Node parent, ArrayList<Token> tokens) {
        super(parent, "ForAssignment", true);
        tokens.remove(0);

        this.variable = new VarExpression(this, tokens);
        this.expression = new Expression(this, tokens);
    }
}

File: Interactive.Genetic.Programming.Environment/src/Domain/Model/Nodes.Big/For/ForAssignment.cs
using Model.Abstract;
using Model.Nodes.Small.Expressions.Standard;

namespace Model.Nodes.Big.For;

public sealed class ForAssignment : Node
{
    public VarExpression Variable { get; }
    public StandardExpression Expression { private get; set; }

    public override List<Node> ChildrenAsNodes()
    {
        var nodes = new List<Node> { this };
        nodes.AddRange(Variable.ChildrenAsNodes());
        nodes.AddRange(Expression.ChildrenAsNodes());
        
        return nodes;
    }

    public ForAssignment(Node parentNode) : base(parentNode, "ForAssignment", true)
    {
        Expression = new StandardExpression(this);
        Variable = new VarExpression(this);
    }
    
    public ForAssignment(Node parentNode, List<Token> tokens) : base(parentNode, "ForAssignment", true)
    {
        tokens.RemoveAt(0);
        Variable = new VarExpression(this, tokens);
        Expression = new StandardExpression(this, tokens);
    }

    public override string ToString() => $"{Variable} = {Expression}";
}